modular_zubbers/modules/plexagon_selfserve/code/plexagon_selfserve.dm

```diff
@@ -0,0 +1,241 @@
+/datum/computer_file/program/crew_self_serve
+	filename = "plexagonselfserve"
+	filedesc = "Plexagon Crew Self Serve"
+	downloader_category = PROGRAM_CATEGORY_SECURITY
+	program_open_overlay = "generic"
+	extended_desc = "Lets crew manage their own HR, giving the impression they have a semblance of control over their employment."
+	program_flags = PROGRAM_ON_NTNET_STORE
+	size = 4
+	tgui_id = "NtosSelfServe"
+	program_icon = "id-card"
+	/// The ID card used to authenticate.
+	var/obj/item/card/id/authenticated_card
+	/// The name of the registered user, related to `authenticated_card`.
+	var/authenticated_user
+	///What trim is applied to inserted IDs?
+	var/target_trim = /datum/id_trim/job/assistant
+
+
+/**
+ * Authenticates the program based on the specific ID card.
+ *
+ * Arguments:
+ * * user - Program's user.
+ * * auth_card - The ID card to attempt to authenticate under.
+ */
+/datum/computer_file/program/crew_self_serve/proc/authenticate(mob/user, obj/item/card/id/id_card)
+	if(isnull(id_card))
+		return FALSE
+
+	authenticated_card = id_card
+	authenticated_user = "[authenticated_card.name]"
+	return TRUE
+
+/datum/computer_file/program/crew_self_serve/on_start(mob/living/user)
+	. = ..()
+	if(!.)
+		return FALSE
+	computer.crew_manifest_update = TRUE
+
+	if(!computer || !computer.computer_id_slot)
+		to_chat(user, span_warning("Plexagon: Login failed, no ID card found!"))
+		playsound(computer, 'sound/machines/buzz-sigh.ogg', 50, FALSE)
+		return FALSE
+	else if(authenticate(user, computer.computer_id_slot))
```

Comment:
I'd pull ```

	authenticated_card = id_card
	authenticated_user = "[authenticated_card.name]"
	return TRUE```
into an unconditional else branch here & remove the authenticate proc. All it does is check if the id_card is null, but you already check !computer.computer_id_slot above.